src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -470,14 +470,18 @@ public static bool IsMountDisabled(this IEnvironment environment)
                 || string.IsNullOrEmpty(environment.GetEnvironmentVariable(MeshInitURI));
         }
 
-        public static CloudName GetCloudName(this IEnvironment environment)
+        public static CloudName GetCloudName(this IEnvironment environment, bool throwOnError = false)
         {
             var cloudName = environment.GetEnvironmentVariable(EnvironmentSettingNames.CloudName);
             if (Enum.TryParse(cloudName, true, out CloudName cloud))
             {
                 return cloud;
             }
 
+            if (throwOnError)
+            {
+                throw new ArgumentException(nameof(GetCloudName), cloudName);
```

Comment:
Consider a different exception type that better conveys the problem. The usage of `ArgumentException` here is not passing the expected parameters